seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/ConfigBuilder.java

```diff
@@ -88,17 +112,30 @@ public static Config of(@NonNull Map<String, Object> objectMap, boolean isEncryp
         return config;
     }
 
-    public static Config of(@NonNull ConfigAdapter configAdapter, @NonNull Path filePath) {
+    public static Config of(
+            @NonNull ConfigAdapter configAdapter, @NonNull Path filePath, List<String> variables) {
         log.info("With config adapter spi {}", configAdapter.getClass().getName());
         try {
             Map<String, Object> flattenedMap = configAdapter.loadConfig(filePath);
             Config config = ConfigFactory.parseMap(flattenedMap);
+            if (variables != null) {
+                Map<String, String> map =
+                        variables.stream()
+                                .filter(Objects::nonNull)
+                                .map(variable -> variable.split("=", 2))
+                                .filter(pair -> pair.length == 2F)
+                                .collect(Collectors.toMap(pair -> pair[0], pair -> pair[1]));
+                config =
+                        config.resolveWith(
+                                ConfigFactory.parseMap(map),
+                                ConfigResolveOptions.defaults().setAllowUnresolved(true));
```

Comment:
Seem like this part show twice in this file. Could you refactor it with new method?

Reply:
Done